pennylane/tape/qscript.py

```diff
@@ -296,17 +297,17 @@ def measurements(self) -> list[MeasurementProcess]:
         return self._measurements
 
     @property
-    def samples_computational_basis(self):
+    def samples_computational_basis(self) -> bool:
         """Determines if any of the measurements are in the computational basis."""
         return any(o.samples_computational_basis for o in self.measurements)
 
     @property
-    def num_params(self):
+    def num_params(self) -> int:
         """Returns the number of trainable parameters on the quantum script."""
         return len(self.trainable_params)
 
     @property
-    def batch_size(self):
+    def batch_size(self) -> Optional[int]:
```

Comment:
Why `Optional`? Can `batch_size` be `None`?

Reply:
I think it's generally `None` if there's no batching, i.e.

```
>>> tape = qml.tape.QuantumScript([qml.RX(1.2, 0)])
>>> tape.batch_size is None
True
```